opentelemetry-stdout/src/logs/exporter.rs

```diff
@@ -1,38 +1,24 @@
 use async_trait::async_trait;
+use chrono::{DateTime, Utc};
 use core::fmt;
-use opentelemetry::{
-    logs::{LogError, LogResult},
-    ExportError,
-};
 use opentelemetry_sdk::export::logs::{ExportResult, LogData};
 use opentelemetry_sdk::Resource;
-use std::borrow::Cow;
-use std::io::{stdout, Write};
+use std::{borrow::Cow, sync::atomic};
 
-type Encoder =
-    Box<dyn Fn(&mut dyn Write, crate::logs::transform::LogData) -> LogResult<()> + Send + Sync>;
-
-/// A [`LogExporter`] that writes to [`Stdout`] or other configured [`Write`].
-///
-/// [`LogExporter`]: opentelemetry_sdk::export::logs::LogExporter
-/// [`Write`]: std::io::Write
-/// [`Stdout`]: std::io::Stdout
+/// An OpenTelemetry exporter that writes Logs to stdout on export.
 pub struct LogExporter {
-    writer: Option<Box<dyn Write + Send + Sync>>,
```

Comment:
I think the good thing about this was it would have allowed users to easily write the output to streams (File, stderr, etc.) other than `stdout`. I would be in favor of keeping the `opentelemetry-stdout` project simple, so I support this change. If there is an actual demand for such an exporter that lets users write to any stream that implements the `Write` trait, we could consider offering such an exporter in the contrib repo.

Reply:
We would need a CHANGELOG entry though as this is a breaking change.

Reply:
There is demand to make an exporter to stdout that can be relied on production. Its not yet spec-ed out, but that is something we should add, once the spec is good. (I think of it as an OTLP Exporter using stdout/file as the transport instead of http/gRPC)

(There is some experimental specs for them I believe)